src/gsHSplines/gsHBox.h

```diff
@@ -14,6 +14,8 @@
 #pragma once
 
 #include <gsIO/gsXml.h>
+#include <gsCore/gsLinearAlgebra.h>
+#include <gsHSplines/gsAABB.h>
```

Comment:
are the new includes needed? 
It is better to minimize inclusions if somehow the compiler finds the definition without them, or even add them to the hpp file instead.
(same consideration for all the new includes in the PR)

Reply:
My reasoning was that it is better to include everything that is used in the file. Otherwise, I cannot guarantee that everything will be fine on the computer of someone else. And I didn't like the prospect of someone getting an error in a code that they did not touch, which can theoretically happen.

With respect to these two particular includes:
- In `gsHBox.h` we need only the declaration of `gsAABB`, its definition is only needed in `gsHBox.hpp`. If you prefer, I could put a forward declaration in `gsHBox.h` and move the include to `gsHBox.hpp`. That way we should be safe while not including too much.
- I added `gsLinearAlgebra` because at some point after touching the order of includes, I got a compilation error complaining about `EIGEN_MAKE_ALIGNED_OPERATOR_NEW` on line 55. That macro is defined somewhere in the linear algebra, thus the include.

Reply:
I agree with the reasoning to include what is used, but minimal inclusions help for better compilation.. Indeed a better way is to forward-declare when possible.
So yes please move the include to gsHBox.hpp
Since linear algebra is needed OK to keep it!

Reply:
Great! I will go through the includes that I touched with this suggestion in mind.

Reply:
Done in [f8edbc](https://github.com/gismo/gismo/pull/679/commits/f8edbc069f2bb0d22fd7a069c40c473205f97e1c), I hope I haven't forgotten anything.